core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -749,6 +763,26 @@ object ZIOSpec extends ZIOBaseSpec {
         )
       } @@ nonFlaky
     ),
+    suite("partitionM")(
+      testM("collect only successes") {
+        val in = List.range(0, 10)
+        for {
+          res <- ZIO.partitionM(in)(a => UIO.succeed(a))
+        } yield assert(res._1, isEmpty) && assert(res._2, equalTo(List.range(0, 10)))
+      },
+      testM("collect only failures") {
+        val in = List.fill(10)(0)
+        for {
+          res <- ZIO.partitionM(in)(a => ZIO.fail(a))
+        } yield assert(res._1, hasSize(equalTo(in.length))) && assert(res._2, isEmpty)
```

Comment:
Can we check that the results contain the expected elements in the correct order in addition to just having the correct size here and in the test below?